core/src/main/java/com/google/cloud/sql/core/DefaultAccessTokenSupplier.java

```diff
@@ -135,6 +134,28 @@ public Optional<AccessToken> get() throws IOException {
     }
   }
 
+  private static boolean isAccessTokenEmpty(GoogleCredentials credentials) {
+    return credentials.getAccessToken() == null
+        || "".equals(credentials.getAccessToken().getTokenValue());
+  }
+
+  private void refreshIfNearExpiration(GoogleCredentials credentials) throws IOException {
+    // if the token does not exist, or if the token expires in less than 4 minutes, refresh it.
+    if (credentials.getAccessToken() == null) {
+      logger.debug("Current IAM Auth Token is not set. Refreshing the token.");
+      credentials.refresh();
+    } else if (credentials.getAccessToken().getExpirationTime() != null
+        && credentials
+            .getAccessToken()
+            .getExpirationTime()
+            .toInstant()
+            .minus(4, ChronoUnit.MINUTES)
```

Comment:
Do we have a constant for this value somewhere? I think we should use it to avoid future mistakes in buffering.

Reply:
➕

Reply:
Good idea, yes. `RefreshCalculator.DEFAULT_REFRESH_BUFFER`